services/api/src/routes/users.js

```diff
@@ -52,6 +54,29 @@ router
       };
     }
   )
+  .post(
+    '/:userId/create-token',
+    requirePermissions({ endpoint: 'users', permission: 'write', scope: 'global' }),
+    async (ctx) => {
+      const { user } = ctx.state;
+
+      // Dont allow an superAdmin to emitate another superAdmin
+      // This should be modified to fit the permissions system required for your project, its somewhat basic in its default state
+      const isViewer = [...user.roles].every((c) => c.role === 'viewer');
```

Comment:
Follow up idea after reading Dom's comment below:

Maybe instead of `allowImpersonation` in `roles.json` we could instead have `impersonationLevel` or something to define the ranking that could drive the logic of whether or not we allow them to impersonate.

To be honest I find that kind of obtuse but I'm throwing it out there as an option as on more complex projects with a variety of roles I'm not really even sure that you can in all cases be able to define which role is "higher" simply by looking at the permissions. I suppose you could assign a ranking to `read` vs `write` and sum them all up but I can easily imagine a situation where you don't want to allow impersonation of user Y by user X even if their roles are technically higher, so maybe it's better to strictly define.

Actually now after thinking that far, what if we had something like:

```
"limitedAdmin": {
  "canImpersonateRoles": [
    "organizer",
    "viewer"
  ]
}
```

Sorry I'm all over the place today